program/rust/src/c_oracle_header.rs

```diff
@@ -34,94 +34,128 @@ pub trait PythAccount: Pod {
     }
 }
 
-impl PythAccount for pc_map_table_t {
+impl PythAccount for MappingAccount {
     const ACCOUNT_TYPE: u32 = PC_ACCTYPE_MAPPING;
-    const INITIAL_SIZE: u32 = PC_MAP_TABLE_T_PROD_OFFSET as u32;
+    const INITIAL_SIZE: u32 = 56; // Tested in test_sizes.rs
 }
 
-impl PythAccount for pc_prod_t {
+impl PythAccount for ProductAccount {
     const ACCOUNT_TYPE: u32 = PC_ACCTYPE_PRODUCT;
-    const INITIAL_SIZE: u32 = size_of::<pc_prod_t>() as u32;
+    const INITIAL_SIZE: u32 = size_of::<ProductAccount>() as u32;
     fn minimum_size() -> usize {
         PC_PROD_ACC_SIZE as usize
     }
 }
 
-impl PythAccount for pc_price_t {
+impl PythAccount for PriceAccount {
     const ACCOUNT_TYPE: u32 = PC_ACCTYPE_PRICE;
-    const INITIAL_SIZE: u32 = PC_PRICE_T_COMP_OFFSET as u32;
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_acc {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_acc {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_map_table {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_map_table {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_prod {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_prod {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price_info {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price_info {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_ema {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_ema {
-}
-
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_pub_key_t {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_pub_key_t {
-}
-
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price_comp_t {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price_comp_t {
-}
+    const INITIAL_SIZE: u32 = 240; // Tested in test_sizes.rs
+}
 
-impl pc_pub_key_t {
-    pub fn new_unique() -> pc_pub_key_t {
+#[repr(C)]
```

Comment:
Trying to not change names too much because some of the fields are used in `upd_aggregate.c`